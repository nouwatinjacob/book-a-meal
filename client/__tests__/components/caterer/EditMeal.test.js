import React from 'react';
import {
  shallow
} from 'enzyme';
import sweetalert from 'sweetalert';
import {
  EditMeal,
  mapStateToProps
} from '../../../src/components/caterer/EditMeal.jsx';

const props = {
  getAMealAction: jest.fn(),
  editMealAction: () => Promise.resolve(),
  mealState: {
    success: true,
    error: {
      response: {
        data: {
          message: 'Error Message'
        }
      }
    }
  },
  match: {
    params: {
      mealdId: 1
    }
  },
  isLoading: true
};

jest.mock('sweetalert');

describe('EditMeal component', () => {
  localStorage.setItem('token', catererToken);
  const setup = () => shallow(<EditMeal {...props} />);

  it('should render EditMeal component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'onFormSubmit'
  when edit meal button is clicked`, (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: jest.fn()
    };

    wrapper.setState({
      mealData: {
        name: 'Fried Rice',
        price: 2600,
        image: 'fried-rice.jpg'
      }
    });

    const onFormSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'onFormSubmit'
    );
    sweetalert.mockResolvedValue(Promise.resolve(true));
    wrapper.instance().onFormSubmit(event);
    expect(onFormSubmitSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'onInputChange'
  when input value is changed`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        name: 'Fried Rice',
        files: ['Fried-rice.jpg']
      }
    };

    const onInputChangeSpy = jest.spyOn(
      wrapper.instance(),
      'onInputChange'
    );
    wrapper.instance().onInputChange(event);
    expect(onInputChangeSpy).toHaveBeenCalled();

    done();
  });
});