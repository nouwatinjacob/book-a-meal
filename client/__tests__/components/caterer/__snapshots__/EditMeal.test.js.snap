// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMeal component should render EditMeal component correctly 1`] = `
<div
  className="container"
>
  <CatererHeader />
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    pauseOnVisibilityChange={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <div
      className="login"
    >
      <div
        className="login-form"
      >
        <h3>
          Edit Meal Details
        </h3>
        <Loader
          color="#ff9600"
          height="50"
          margin="2px"
          type="Rings"
          visible={true}
          width="100"
        />
        <form
          encType="multipart/form-data"
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder=""
            type="text"
          />
          <br />
          <input
            name="price"
            onChange={[Function]}
            placeholder=""
            type="text"
          />
          <br />
          <input
            name="image"
            onChange={[Function]}
            placeholder="Meal Image"
            type="file"
          />
          <br />
          <button
            className="button warning"
          >
            Modify Meal
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
